packages/market-service/src/market-service-manager.ts

```diff
@@ -58,16 +58,62 @@ export class MarketServiceManager {
 
   async findAll(args: FindAllMarketArgs): Promise<MarketCapResult> {
     let result: MarketCapResult | null = null
-    // Go through market providers listed above and look for market data for all assets.
-    // Once data is found, exit the loop and return result. If no data is found for any
-    // provider, throw an error.
+    if (args.fetchFromAllProviders) {
+      /** Some of the providers above have performance issues. Instead of modifying the default behavior,
+       * we'll just use a blacklist exclude the questionable providers with the fetchFromAllProviders option is enabled.
+       */
+      const blacklistedProviders = [
```

Comment:
lemme provide a little context and potentially simpler strategy here

iterating the `marketProviders` array in this `findAll` method never actually made any sense.

from the PR description i'm gathering the intent of this is to get market data for osmo tokens.

if that's the case, this *feels* like the responsibility of the defi resolvers in the front end, as this market data is only required for that domain of the app, and we could just call the `findByAssetId` method with each required asset id. if that's too noisy, or we agree we kind of always need this data, we could do this here with just a whitelist of coingecko + osmo data providers.

what do you reckon?

Reply:
after discussing - root cause/motivation of this PR was market data unable for long tail ibc tokens at runtime

IMO, the respective domain (osmo LPing) in the client should be responsible for fetching this lazily as required using the `findByAssetId` method, which can be called with an automatically generated RTK query hook (`useFindByAssetIdQuery`) or imperatively, like so https://redux-toolkit.js.org/rtk-query/api/created-api/endpoints#example-1